cli/azd/pkg/osutil/retry_strategy.go

```diff
@@ -0,0 +1,37 @@
+package osutil
+
+import (
+	"os"
+	"strings"
+	"time"
+)
+
+type RetryStrategy struct {
+	// The maximum number of retries before failing
+	MaxRetries uint64
+	// The time between each retry attempt
+	RetryBackoff time.Duration
+}
+
+// Creates a new retry strategy that also reduces the time for when running in test
+func NewRetryStrategy(maxRetries uint64, retryBackoff time.Duration) *RetryStrategy {
```

Comment:
Do we really need this? In how many places is this facility used?

Reply:
Used in a few places right now and its nice to have it in a consolidated place to start.